### PR TITLE
refactor: Full re-write of SDK

### PR DESCRIPTION
**Pull request by Zambrella**
**- What I did**
I re-wrote the code from scratch to allow for better extensibility and maintainability (and for fun). There is no additional functionality (yet) from the previous version. The main changes include:
- Breaking the repo down into [workspaces](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html)
- Created traits for crypto and TLS to allow "drop-in" replacements if needed in the future
- Used `RustCrypto` crates for crypto because it is written in pure Rust so should be compatible with more hardware
- Used `rustls` crate for TLS which should have broader compatibility that the previous crate that was being used
- More tests

**- How to verify it**
- Check through the code
- Run the tests (info about how to do this in the `README.md`)
- Run the examples (info about how to do this in the `README.md`) and check the data is read and written correctly and can be accessed by other SDKs

**- Description for the changelog**
Full re-write of SDK

### CODE REVIEW

**Comment by cpswan** [comment]
Thanks @Zambrella 

Can you please take a look at the clippy warnings and fix them up (you might even be able to use it to implement the suggested fixes).

**Comment by Zambrella** [comment]
@cpswan Clippy warnings have been fixed :)

File: Cargo.toml
Comment (cpswan) [inline Cargo.toml:12]: Please bump the version to 0.3.0

File: at_chops/test_data/test_rsa_private_key.der
Comment (cpswan) [inline at_chops/test_data/test_rsa_private_key.der:1]: We really shouldn't have binaries in the repo. Could these be substituted with text encoded keys?

**Review by cpswan — APPROVED** [review]
@Zambrella thanks for this. It's a brilliant contribution :)

Secondly my apologies that it's taken me so long to review. It's a big PR and I wanted the chance to properly test it. It took some travel disruption and an unexpected change to my weekend to clear a bit of time.

There are a few minor tweaks I'd love to have before merging, but if you don't have the bandwidth to come back to this right now then that's entirely understandable. So I'm approving and will merge anyway in a week or so if I don't hear from you.

**Pull request merged**
